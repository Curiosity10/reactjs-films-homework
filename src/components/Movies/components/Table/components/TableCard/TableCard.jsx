import React from 'react';
import PropTypes from 'prop-types';
import styles from './TableCard.scss';
import NumRating from '../../../../../NumRating';
import Genres from '../../../../../Genres';
import Button from '../../../../../Button';
import StarRating from '../../../../../StarRating';
import truncate from '../../../../../../utils/truncate';

const TableCard = ({ movie }) => {
  const {
    rating,
    title,
    genres,
    overview,
    image,
  } = movie;
  const maxLength = 500;
  return (
    <article className={styles.card}>
      <img src={image} alt="Poster" />
      <div className={styles.info}>
        <h2 className={styles.title}>{title}</h2>
        <Genres genres={genres} />
        <p className={styles.overview}>{truncate(overview, maxLength)}</p>
        <div className={styles.rating}>
          <StarRating starsSelected={rating} />
          <NumRating rating={rating} />
        </div>
        <Button type="primary">Watch Now</Button>
      </div>
    </article>
  );
};

TableCard.propTypes = {
  movie: PropTypes.shape({
    rating: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    genres: PropTypes.array.isRequired,
    overview: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
  }).isRequired,
};

export default TableCard;
