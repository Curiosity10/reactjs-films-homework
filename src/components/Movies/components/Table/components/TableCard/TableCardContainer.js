import { connect } from 'react-redux';
import TableCard from './TableCard';
import { fetchVideoKey } from '../../../../../../modules/app/appActions';

export default connect(null, { fetchVideoKey })(TableCard);
