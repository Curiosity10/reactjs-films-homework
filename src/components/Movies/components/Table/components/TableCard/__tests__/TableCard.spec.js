import React from 'react';
import { create, act } from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';
import TableCard from '../TableCard';

describe('TableCard component renders correctly', () => {
  const movie = {
    title: 'test',
    id: 1,
    genre_ids: [
      {
        id: 12,
        name: 'Action',
      },
      {
        id: 16,
        name: 'Adventure',
      },
    ],
    overview: 'desc',
  };

  it('TableCard renders correctly', () => {
    const tree = create(
      <StaticRouter location="someLocation">
        <TableCard
          movie={movie}
        />
      </StaticRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('TableCard watchBtnClick works correctly', () => {
    const tree = create(
      <StaticRouter location="someLocation">
        <TableCard
          movie={movie}
        />
      </StaticRouter>,
    );

    const watchBtn = tree.root.findByProps({ 'aria-label': 'Watch Now' });
    const mockWatchBtn = jest.fn(watchBtn.props.onClick);

    act(() => {
      mockWatchBtn();
    });

    expect(mockWatchBtn).toHaveBeenCalled();
  });
});
