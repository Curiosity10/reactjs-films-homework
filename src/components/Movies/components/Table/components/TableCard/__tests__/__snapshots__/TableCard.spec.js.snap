// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCard component renders correctly TableCard renders correctly 1`] = `
<article
  className="card"
>
  <img
    alt="Poster"
  />
  <div
    className="info"
  >
    <h2
      className="title"
    >
      <a
        href="/movie/:1"
        onClick={[Function]}
      >
        test
      </a>
    </h2>
    <p
      className="overview"
    >
      desc
    </p>
    <div
      className="rating"
    />
    <button
      aria-label="Watch Now"
      className="btn primary"
      onClick={[Function]}
      type="button"
    >
      Watch Now
    </button>
  </div>
</article>
`;
